### PR TITLE
MAINT: Fix warnings

### PR DESCRIPTION
**Pull request by bashtage**
Fix many of the large number of  Future/PendingDeprecation warnings that Pytest 3.8 now shows.

### CODE REVIEW

**Comment by bashtage** [comment]
xref #5132

**Comment by bashtage** [comment]
@josef-pkt @jbrockmendel This is ready now.  It fixes all future/pending dep/etc warnings and silences all of the upstream generated warnings (tuple access in scipy and mkl_fft, categorical in patsy).

**Comment by josef-pkt** [comment]
LGTM based on quick skimming
will look again and merge later today

**Comment by bashtage** [comment]
It will match any warning starting with 'matrix subclass'

**Comment by josef-pkt** [comment]
I'm merging this without waiting for appveyor, so we get Travis green again.

looks good, 
corrpsd might need a separate look for why we use `matrix` (but that could come also from the use of scipy.sparse)

@bashtage Thank you

**Comment by bashtage** [comment]
It is all from sparse.

File: statsmodels/tsa/statespace/tools.py [hunk 1]
```diff
@@ -1799,7 +1799,7 @@ def prepare_trend_spec(trend):
     if trend is None or trend in ['n', 'nc']:
         polynomial_trend = np.ones((0))
         if trend == 'nc':
-            warnings.warn("Argument option trend='nc' is deprecated. Please"
+            warnings.warn("Argument option trend='n' is deprecated. Please"
                           " use option trend='n'.", DeprecationWarning)
```
Comment (jbrockmendel) [inline statsmodels/tsa/statespace/tools.py]: Does this message make sense?

**Pull request merged**